webpages/register.html

```diff
@@ -0,0 +1,139 @@
+<!doctype html>
+<title>Username | Living Meta-Analysis</title>
+<script src="/lib/promise.js"></script>
+<script src="/lib/fetch.js"></script>
+<script src="https://apis.google.com/js/platform.js"></script>
+<link rel="stylesheet" href="/css/main.css">
+
+<header>
+  <a href="/" id="logo-box">
+    <h1>LiMA</h1><h2>Living Meta-Analysis</h2>
+  </a>
+</header>
+
+<h1>Please choose a username:</h1>
+<div class="register">
+  <p class="edithighlight">
+    <span class='customusername' contenteditable placeholder="custom username"></span>
+  </p>
+
+  <form class='suggestions'>
+  <h3>Suggestions:</h3>
+    <span id='emailaddress'> same as email address</span>
+    <span id='emailaddressfirst'> same as email address first part</span>
+    <span id='displayname'> same as Google display name (without spaces)</span>
+  <p>
+    <button class="confirm">confirm</button>
+  </p>
+  <p class="output"></p>
+</div>
+
+<footer>
+  <p>LiMA (Living Meta-Analysis) at <a href="http://port.ac.uk/">University of Portsmouth</a>, &copy; 2016–2017</p>
+  <p>Feedback and questions are welcome at <a href="mailto:lima@port.ac.uk">lima@port.ac.uk</a>.</p>
+</footer>
+
+
+<script src="/js/tools.js"></script>
+<script src="/js/auth.js"></script>
+<script>
+  'use strict';
+  var lima = window.lima;
+  var _ = lima._;  // underscore symbol is used for brevity, defined in tools.js
+
+  /^[a-zA-Z0-9-_]+$/;
+  var USERNAME_RE = /^[a-zA-Z][a-zA-Z0-9-_@\.]*$/;
+
+  window.lima.initPage = function () {
+
+    var email = lima.getAuthenticatedUserEmail();
+
+    if (!email) {
+      window.location.href = '/signin';
+    }
+
+    var currUser;
+
+    lima.getGapiIDToken()
+      .then(function (idToken) {
+        return fetch('/api/profile/' + email, _.idTokenToFetchOptions(idToken));
+      })
+      .then(_.fetchJson)
+      .then(function (user) {
+        currUser = user;
+      })
+      .catch(function (err) {
+        if (err.status === 404) {
+          console.warn("Possibly this user isn't registered");
+        }
+      });
+
+      if (!currUser) {
+        // We don't have that user, so just use google's..
+        currUser = gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile();
```

Comment:
We dont need to get this here, we can use the functions built into the returned object...

e.g.

```
gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile().getEmail()
```